PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp

```diff
@@ -17,24 +17,39 @@
 
 #include <shobjidl.h>
 #include <shlwapi.h>	// PathIsDirectory
-#include <comip.h>		// _com_ptr_t
+#ifdef __MINGW32__
+#include <cwchar>
+#endif
 #include <comdef.h>		// _com_error
+#include <comip.h>		// _com_ptr_t
 
 #include "CustomFileDialog.h"
 #include "Parameters.h"
 
+// Workaround for MinGW because its implementation of __uuidof is different.
+template<class T>
+struct ComTraits
+{
+	static const GUID uid;
+};
+template<class T>
+const GUID ComTraits<T>::uid = __uuidof(T);
+
 // Smart pointer alias for COM objects that makes reference counting easier.
 template<class T>
-using com_ptr = _com_ptr_t<_com_IIID<T, &__uuidof(T)>>;
+using com_ptr = _com_ptr_t<_com_IIID<T, &ComTraits<T>::uid>>;
 
 // _com_ptr_t needs this function.
 // Define it here to avoid linking with "comsupp.lib".
+// MinGW defines its own version of this function.
```

Comment:
Is there a downside to linking with comsupp.lib ?

Reply:
No, I am not aware of any.
It just seemed like overkill to include a .lib just for a single simple function.
Do you think we can add it and let the optimizer do the thing?

Reply:
To me a dozen characters added to the link specification beats a bunch of source lines of conditional code.
Not sure what @donho prefers.

Reply:
@mere-human 
How can it "overkill" to link a .lib? 
@sasumner is right. Let's link with comsupp.lib - I prefer to include comsupp.lib in the project file than including a bunch of source lines which is not belong to us (and I don't think we have the right - since it's not under GPL).